packages/loader/container-loader/src/protocol.ts

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IAudienceWriter } from "@fluidframework/container-definitions";
+import {
+    ILocalSequencedClient,
+    IProtocolHandler as IBaseProtocolHandler,
+    IQuorumSnapshot,
+    ProtocolOpHandler,
+} from "@fluidframework/protocol-base";
+import {
+    IDocumentAttributes,
+    IProcessMessageResult,
+    ISequencedDocumentMessage,
+    ISignalClient,
+    ISignalMessage,
+    MessageType,
+} from "@fluidframework/protocol-definitions";
+
+export interface IProtocolDetails {
+    /**
+     * Function to be used for creating a protocol handler.
+     */
+    protocolHandlerBuilder: ProtocolHandlerBuilder;
```

Comment:
Do you expect more members to be added to this interface? If no, maybe we do not need IProtocolDetails interface - can work directly with ProtocolHandlerBuilder

Reply:
The point of it was to keep both the builder and the audience but then the audience got removed. Afterwards, I was thinking that in the exceptional case of needing something extra here, it would make the migration easier. Thanks for calling it out, I will remove it. The API change would be bad either way and we can make a holder interface then.